plugins/aws/resoto_plugin_aws/resource/s3.py

```diff
@@ -20,18 +20,16 @@ class AwsS3Bucket(AwsResource, BaseBucket):
 
     @classmethod
     def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
-        def add_tags(bucket: AwsS3Bucket) -> None:
-            tags = builder.client.call(
-                service="s3", action="get_bucket_tagging", result_name="TagSet", Bucket=bucket.name
-            )
+        def add_tags(bucket: AwsS3Bucket, client: AwsClient) -> None:
+            tags = bucket._get_tags(client)
             if tags:
-                bucket.tags = bend(ToDict(), tags)
+                bucket.tags = cast(Dict[str, Optional[str]], tags)
```

Comment:
WDYT about changing the return type of the _get_tags method instead of casting it to the right type?

Reply:
Sure, current return type is `Dict[str, str]` which seems very compatible to `Dict[str, Optional[str]]`. That said, in AWS there's always a value, so `Optional[str]` seems wrong?!

Reply:
this would mean changing all the types for the follow up methods like `set_tags` too. And an optional value in `set_tags` specifically doesn't seem right 🤔 Either way this is the more minimal change